core/src/main/java/org/jenkins/ui/icon/IconSet.java

```diff
@@ -92,7 +99,7 @@ public static String getSymbol(String name, String title, String tooltip, String
         }
 
         // Load symbol if it exists
-        InputStream inputStream = IconSet.class.getResourceAsStream("/images/symbols/" + translatedName + ".svg");
+        InputStream inputStream = getClassLoader(identifier).getResourceAsStream("/images/symbols/" + translatedName + ".svg");
```

Comment:
this isn't working. 😢 

I just get null when trying to use the plugins classloader that the plugin manager holds.

@janfaracik 's original design of passing the calling class won't work as we would want to have things like credentials plugin supplying icons that other plugins use, and library plugins like font-awesome, so we need to use the appropriate plugin classloader.

@jglick or @daniel-beck any idea what I'm doing wrong here?
Example plugin using this change: https://github.com/janfaracik/ui-samples-plugin/tree/overhaul

(core icons load fine as that just falls back to IconSet classloader)

Reply:
Are you running inside a test or `hpi:run`?

Reply:
`hpi:run`, this seems to fix it [`3137a98` (#6395)](https://github.com/jenkinsci/jenkins/pull/6395/commits/3137a98b7124b397d5aad1f26aa3ee313bdc6e72)